### DISCUSSION TITLE
Modifying the Aeotec Doorbell 6 config file

### DISCUSSION THREAD

**Discussion by Matt-PMCT** [zwave-js/zwave-js#2542] [General] [answered]
I have an Aeotec Doorbell 6 and I noticed that interviewing takes forever because it has to get every single tone name 21 times. I have attached the log file while I did a re-interview (Node 7). For example if you search for "ding-dong" you'll see it occurs 21 times during the interview. I was wondering about the following things:

1.) If I create the list of sounds and add to config file would it help anything?

2.) There is also things missing from the Notification section, for example there is a "Notification" to 112-3-3-Siren_Status that is sent when the doorbell is pushed, but after a re-interview that object is not there/missing. Is this also something I should add to the config file?

3.) At the beggining of the interview I see a message saying " [Node 007]   does NOT support CC Battery (0x80) ". Then later in the interview there are a bunch of requests for BatteryCC options that fail and timeout. Is that an issue I should submit?

Basically wondering if the config files is the right place to add this stuff to see if I can be helpful by submitting a PR.

[zwavejs_1.log](https://github.com/zwave-js/node-zwave-js/files/6424945/zwavejs_1.log)

**Comment by AlCalzone** [comment, marked as answer]
> interviewing takes forever

First off - 5 minutes is not forever and well within the 10 minutes the specifications mention. Also it should only be necessary once. However, we can do better.

> For example if you search for "ding-dong" you'll see it occurs 21 times during the interview

The device has 8 endpoints and each of them supports Sound Switch CC with 30 tones. To my knowledge there's no way to know that endpoint 8 has the same sounds as endpoint 1 without querying all of them. However I don't think this causes the long interview.

> 1.) If I create the list of sounds and add to config file would it help anything?

No.

> [Node 007] does NOT support CC Battery (0x80)

I think that comes from the device class definitions which don't distinguish between always listening and sleeping nodes for the mandatory battery support. All we could do in a device config file is a workaround, but I'd rather fix this at the root.
This should amount to ~90s of wasted time alone.

Then there are another 8 queries (~80s) for `Manufacturer Specific CC` on the endpoints, which again comes from the chosen device class. I'll need to read up if the device does something wrong here. If it does, we can skip the query for the endpoints by force-removing it from the list of supported CCs.

**Reply by AlCalzone** [reply]
I've raised https://github.com/zwave-js/node-zwave-js/issues/2543 to track this.
